### PR TITLE
feat(lsp): add support for astro, vue and svelte

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR adds partial support for Astro, Vue and Svelte files in the LSP.

In order to achieve that, I had to apply some workarounds and some required changes.

### Required changes

So far we were retrieving the file source of a file by using their `AnyParse`, which is stored inside the Workspace. This methodology was "naive" and imperfect, and it showed its flaws in this feature.

The file source was recreated using their file path, but unfortunately this is no longer true due to the nature of `svelte` and `vue` files, which support multiple languages regardless of their extension. They do that by using the `lang` attribute of the `<script>` element.

So I took a different approach. I removed the concept of `AnyFileSource`. The function `parse` now returns a new file hint, which is computed during the parsing phase. This new file hint is stored inside the struct `Document`. If the parsing phase returns a file hint, we update the document with this one.

Then, I updated some function calls to accept this language hint. We use this language hint to create the correct file source.

### Workarounds

The workarounds are all inside the LSP, and they consist in changing the position of the cursor and the position of the diagnostics. This is necessary because when we store a parsed result inside the workspace, we only store the JavaScript portion of the file, but when working with LSP, they have the positions of the whole file.

The "handler" files now have a function called `offset`, which returns the byte offset position of their initial fence, and we create a new `TextRange` by adding this new offset.

By applying this workaround, we can now signal the correct position of the diagnostics and the code actions.

## Test Plan

I created a new test. And the current CI should pass.

### CODE REVIEW

File: crates/biome_service/src/file_handlers/svelte.rs [hunk 1]
```diff
@@ -31,6 +33,52 @@ lazy_static! {
     .unwrap();
 }
 
+impl SvelteFileHandler {
+    // It extracts the JavaScript/TypeScript code contained in the script block of a Vue file
```
Comment (panaman67) [inline crates/biome_service/src/file_handlers/svelte.rs:37]:
```suggestion
    // It extracts the JavaScript/TypeScript code contained in the script block of a Svelte file
```

File: crates/biome_lsp/Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,7 @@ serde_json         = { workspace = true }
 tokio              = { workspace = true, features = ["rt", "io-std"] }
 tower-lsp          = { version = "0.19.0" }
 tracing            = { workspace = true, features = ["attributes"] }
+log = "0.4.20"
```
Comment (anonrig) [inline crates/biome_lsp/Cargo.toml:35]: Should this be workspace?

File: crates/biome_lsp/src/handlers/formatting.rs [hunk 3]
```diff
@@ -28,9 +29,36 @@ pub(crate) fn format(
...
+        })?;
+
+        let mut output = printed.into_code();
+        let file_extension = rome_path.extension().and_then(|s| s.to_str());
+        let input = session.workspace.get_file_content(GetFileContentParams {
+            path: rome_path.clone(),
+        })?;
+        match file_extension {
+            Some("astro") => {
+                if output.is_empty() {
```
Comment (anonrig) [inline crates/biome_lsp/src/handlers/formatting.rs:43]: We can move this check before match statement

File: crates/biome_service/src/file_handlers/svelte.rs [hunk 4]
```diff
@@ -31,6 +33,55 @@ lazy_static! {
     .unwrap();
 }
 
+impl SvelteFileHandler {
+    // It extracts the JavaScript/TypeScript code contained in the script block of a Svelte file
```
Comment (anonrig) [inline crates/biome_service/src/file_handlers/svelte.rs:37]:
```suggestion
    /// It extracts the JavaScript/TypeScript code contained in the script block of a Svelte file
```

**Review by anonrig — APPROVED** [review]

**Pull request merged**